.github/workflows/unittests.yaml

```diff
@@ -29,27 +31,18 @@ jobs:
           lfs: true
       - name: Checkout LFS objects
         run: git lfs checkout        
-      - name: Setup Scala
-        uses: olafurpg/setup-scala@v12
-        with:
-          java-version: adopt-openj9@1.8.0-292
-      - name: Cache .ivy2
-        uses: actions/cache@v4
-        with:
-          path: ~/.ivy2
-          key: ${{ runner.os }}-ivy2-${{ hashFiles('**/build.sbt') }}
-      - name: Cache .sbt
-        uses: actions/cache@v4
+      - name: Setup cache
+        uses: coursier/cache-action@v6
+      - name: setup Scala
+        uses: coursier/setup-action@v1
```

Comment:
suggestion: while we're here, shall we start the practice of using git hashes for versions instead?